Source/Interactions/Cursor_input.cpp

```diff
@@ -18,11 +18,11 @@ void cursor_position_callback(GLFWwindow* window, double xpos, double ypos)
 
 void mouse_button_callback(GLFWwindow* window, int button, int action, int mods)
 {
-    if (MainMenu->IS_MENU_ACTIVE) {
-        Log.error("MainMenu");
-        MainMenu->cursor_click(button);
-    } else if (PauseMenu->IS_MENU_ACTIVE) {
-        Log.error("PauseMenu");
-        PauseMenu->cursor_click(button);
+    if (action == GLFW_PRESS) {
+        if (MainMenu->IS_MENU_ACTIVE) {
```

Comment:
So Ideally, rather than checking if main menu is active every single time we press a mouse button. We can instead have it so that when the main menu is active we set the mouse callback to one that only handles the main menu stuff. Then when the main menu is not active we can set the mouse call back to a different function. We can then even make the callback local to the object, that way we don't need to expose the main menu pointer to other files